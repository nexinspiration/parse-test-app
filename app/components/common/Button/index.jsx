import React from 'react';

import styles from '../../../styles/app.css';

export default function Button(props) {

  const handleKeyPress = (event) => {
    // On Enter Press (13), consider it as Button Click
    if (event.which === 13) {
      props.onClick();
    }
  }

  return (
    <button
      className={styles.btn}
      onClick={() => {props.onClick()}}
      onKeyPress={(event) => {handleKeyPress(event)}}>
      {props.text || 'Execute'}
    </button>
  );
}