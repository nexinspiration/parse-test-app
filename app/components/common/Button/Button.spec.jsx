import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from '../Button';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

describe('Button', () => {
  it('should return an element.', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.containsMatchingElement(<button>Execute</button>)).toEqual(true);
  });

  // Event tests
  it('Button action triggered on Enter Press', () => {
    const buttonClick = sinon.spy();
    const wrapper = mount(
      <Button onClick={buttonClick}/>
    );
    wrapper.find('button').simulate('keypress', { which: 13 });
    expect(buttonClick.calledOnce).toEqual(true);
  });

  it('Button action triggered on Click', () => {
    const buttonClick = sinon.spy();
    const wrapper = mount(
      <Button onClick={buttonClick}/>
    );
    wrapper.find('button').simulate('click');
    expect(buttonClick.calledOnce).toEqual(true);
  });

  // Snapshot tests
  it('Renders Correctly', () => {
    const tree = renderer.create(
      <Button/>
    );
    expect(tree).toMatchSnapshot();
  });

  it('Renders Passed Text Correctly', () => {
    const tree = renderer.create(
      <Button text='Submit'/>
    );
    expect(tree).toMatchSnapshot();
  });
});