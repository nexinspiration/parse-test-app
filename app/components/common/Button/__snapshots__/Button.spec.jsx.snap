// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders Correctly 1`] = `
<button
  className={undefined}
  onClick={[Function]}
  onKeyPress={[Function]}
>
  Execute
</button>
`;

exports[`Button Renders Passed Text Correctly 1`] = `
<button
  className={undefined}
  onClick={[Function]}
  onKeyPress={[Function]}
>
  Submit
</button>
`;
